utils/post_utils.jsx

```diff
@@ -93,6 +93,38 @@ export function canEditPost(post, editDisableAction) {
     return canEdit;
 }
 
+export function canRethreadPost(post, editDisableAction) {
+    if (isSystemMessage(post)) {
+        return false;
+    }
+
+    let canRethread = false;
+    let canRethreadOthers = false;
+    const license = getLicense(store.getState());
+    const config = getConfig(store.getState());
+    const channel = getChannel(store.getState(), post.channel_id);
+
+    const isOwner = isPostOwner(post);
+    canRethread = haveIChannelPermission(store.getState(), {channel: post.channel_id, team: channel && channel.team_id, permission: Permissions.EDIT_POST});
+    canRethreadOthers = haveIChannelPermission(store.getState(), {channel: post.channel_id, team: channel && channel.team_id, permission: Permissions.EDIT_OTHERS_POSTS});
```

Comment:
does it really make sense to use the "edit other posts" permission for this? as someone running a mattermost server I would probably NOT allow (m)any people  to edit others' posts, but allowing rethreading is much less invasive and something i'd probably let every team admin do, or even a channel admin.

to me it sounds like having a separate permission for this would be better UX, especially for the "others" version

Reply:
This needs more discussion in my opinion. The current solution "cuts corners" by using existing edit_post permission,  but I'm not entirely sure how to add new permissions and integrate them to the permission system/scheme etc.